In [3]:
import pandas as pd

BASE = '../datasets/HR-Employee-Attrition.csv'

df = pd.read_csv(BASE)

### Introdução a variáveis categóricas:

Variáveis categóricas são em outras palavras colunas de um DataSet com pouca variação de valores. Alguns profissionais da área de ciência de dados chamam de variáveis discretas por que tendem a manter valores bem especificos de pouco dinamismo. Essas variáveis são úteis para agruparmos um conjunto de registros dentro de uma única categoria.

In [20]:
# Vamos verificar o quão variante são os valores de cada coluna do nosso DataFrame
#e entender a partir deles quais podemos utilizar como variáveis categóricas.

for c in df.columns:
    print(c, df[c].nunique())
    print()

Age 43

Attrition 2

BusinessTravel 3

DailyRate 886

Department 3

DistanceFromHome 29

Education 5

EducationField 6

EmployeeCount 1

EmployeeNumber 1470

EnvironmentSatisfaction 4

Gender 2

HourlyRate 71

JobInvolvement 4

JobLevel 5

JobRole 9

JobSatisfaction 4

MaritalStatus 3

MonthlyIncome 1349

MonthlyRate 1427

NumCompaniesWorked 10

Over18 1

OverTime 2

PercentSalaryHike 15

PerformanceRating 2

RelationshipSatisfaction 4

StandardHours 1

StockOptionLevel 4

TotalWorkingYears 40

TrainingTimesLastYear 7

WorkLifeBalance 4

YearsAtCompany 37

YearsInCurrentRole 19

YearsSinceLastPromotion 16

YearsWithCurrManager 18



### O que é preciso para uma variável ser considerada como categórica? 

É possível identificar se uma coluna pode ou não ser classificada como uma "variável categórica" pela variância de seus valores e neste caso devemos considerar que uma grande variação dos dados não nos permite considerar esta coluna como categórica, e também pelo propósito que aquela coluna tem no nosso Dataset, ou seja, não basta ter uma pequena variação de dados é necessário também entendermos se aqueles dados podem de fato categorizar os registros do nosso Dataset. Inclusive analisar a semântica de uma variável para determina-la como categórica é mais determinante do que analisar o quão variante é seus dados.

Um exemplo, a coluna "Age" possui 43 variações, um número relativamente baixo se comparado com o total de registros (1470), ou seja, quantitativamente falando é "ok" determina-la como categórica, mas trata-se de um dado muito dinâmico, ou seja, estamos falando de uma variável na qual os dados pertencentes a uma certa categoria vão pertencer depois de no máximo 365 dias e isso contraria o conceito de categorização.  

É importante ressaltar, valores de poucas variações e não mutáveis, em resumo, variáveis discretas.